riscv/mmu.h

```diff
@@ -187,6 +211,21 @@ class mmu_t
     })
   }
 
+  // for shadow stack amoswap
+  template<typename T>
+  T ssamoswap(reg_t addr, reg_t value) {
+    convert_load_traps_to_store_traps({
+      bool forced_virt = false;
+      bool hlvx = false;
+      bool lr = false;
+      bool ss_access = true;
+      store_slow_path(addr, sizeof(T), nullptr, {forced_virt, hlvx, lr, ss_access}, false, true);
+      auto return_address = ss_load<T>(addr);
+      ss_store<T>(addr, value);
+      return return_address;
```

Comment:
I think we should use `load` and `store` with `ss_access=true` in `ssamoswap` instead of `ss_load` and `ss_store`. The `ss_load` and `ss_store` will cause store-access-fault if the address is misaligned which is right for `sspush/sspopchk` but for `ssamoswap` the misaligned address should caused either a) address-misaligned exception or b) access-fault exception as done by `store_slow_path`.

Reply:
Thanks, I will replace ss_load/ss_store with load/stoe and ss_access flags is true